src/main/cpp/util/md5.cc

```diff
@@ -206,10 +207,18 @@ void Md5Digest::Transform(
   // ROTATE_LEFT rotates x left n bits.
 #define ROTATE_LEFT(x, n) (((x) << (n)) | ((x) >> (32-(n))))
 
+// SET reads 4 input bytes in little-endian byte order and stores them
```

Comment:
This appears to be a regression for little endian.
I think this should be defined differently for big vs. little. 
But,  I still think it can be better than that.  I think you can do better by using x_pos in each but incrementing it after line 224.

Reply:
Hi @aiuto, 

Thanks for checking! However, I don't think it will be a regression for little-endian because it stores the data in host byte order.

Reply:
This replaces a single 4 byte load&store with many more load operations. The compiler *might* be smart enough to reduce it all, but I wouldn't count on it.

Perhaps something like:
`.... + (*x_pos++ = HOST_TO_LE32(*curword++))`

And 
```
#if little_endian
#define HOST_TO_LE32(n) (n)
#else
#define HOST_TO_LE32(n) ((n)>>24 & 0x000000ff) | ((n)>>8) & 0x0000ff00) |  ((n)<<8) & 0x00ff0000) | ((n) << 24) & 0xff000000))
#endif 
```

Reply:
Yeah, makes sense, a single `le32toh()` function can do the same thing.

Reply:
Hi @aiuto ,

I've updated the code to use `le32toh()` function and the md5 test passed on both little-endian and big-endian systems. Could you please take a look? Thanks!